tests/rust/src/bin/stdio.rs

```diff
@@ -1,13 +1,47 @@
-use wasi_tests::{STDERR_FD, STDIN_FD, STDOUT_FD};
+use std::{env, process};
 
-unsafe fn test_stdio() {
-    for fd in &[STDIN_FD, STDOUT_FD, STDERR_FD] {
-        wasi::fd_fdstat_get(*fd).expect("fd_fdstat_get on stdio");
-        wasi::fd_renumber(*fd, *fd + 100).expect("renumbering stdio");
+use wasi_tests::{create_tmp_dir, open_scratch_directory, STDERR_FD, STDIN_FD, STDOUT_FD};
+
+unsafe fn test_stdio(dir_fd: wasi::Fd) {
+    for stdio_from_fd in &[STDIN_FD, STDOUT_FD, STDERR_FD] {
+        let stdio_to_fd =
+            wasi::path_open(dir_fd, 0, "file", wasi::OFLAGS_CREAT, 0, 0, 0).expect("open file");
+
+        wasi::fd_renumber(*stdio_from_fd, stdio_to_fd).expect("renumbering stdio");
+
+        wasi::fd_fdstat_get(stdio_to_fd).expect("fd_fdstat_get failed");
+        wasi::fd_fdstat_get(*stdio_from_fd).expect_err("stdio_from_fd is not closed");
```

Comment:
Is it expected that the source fd is being closed in that case?

Reply:
Yes, that's the behaviour that's currently enforced by the [renumber](https://github.com/WebAssembly/wasi-testsuite/blob/main/tests/rust/src/bin/renumber.rs#L49-L52) test. The docs don't explicitly mention closing the source fd, but `fd_renumber` is meant to be analogous (with a few differences) to [dup2](https://pubs.opengroup.org/onlinepubs/009604499/functions/dup.html), which states that the source fd will be closed.